linkcheck/listeners.py

```diff
@@ -21,7 +23,12 @@ def linkcheck_worker():
     global worker_running
     while tasks_queue.not_empty:
         task = tasks_queue.get()
-        task['target'](*task['args'], **task['kwargs'])
+        # An error in any task should not stop the worker from continuing with the queue
+        try:
+            task['target'](*task['args'], **task['kwargs'])
+        except:  # pylint: disable=bare-except
+            logger.error("An error occured while checking links: %s",
+                         repr((task['args'], task['kwargs'])))
```

Comment:
```suggestion
        except Exception as e:
            logger.exception(
                "%s while running %s with args=%r and kwargs=%r: %s",
                type(e).__name__,
                task['target'].__name__,
                task['args'],
                task['kwargs'],
                e
            )
```
Maybe we could log the whole stack trace here to find out what exactly went wrong?
(also, small typo at "occured" and `repr` can be omitted when using the `%r` formatter)

Reply:
Thanks

Reply:
@claudep we could add a stack trace only if the Django Debug mode is enabled?